cs/ccxt/base/Exchange.Options.cs

```diff
@@ -254,6 +254,7 @@ void initializeProperties(dict userConfig = null)
             var concurrentExtendedDict = new ConcurrentDictionary<string, object>(extendedDict);
             this.options = concurrentExtendedDict;
         }
+        this.options = this.deepExtend(this.getDefaultOptions(), this.options);
```

Comment:
if you do this using deepDeepExtend options will not be concurrent-safe

Reply:
yes i had drafted this PR, but atm i've pushed a fix, this should be ok now.

Reply:
ready now